lib/find_non_repeated_letter.rb

```diff
@@ -9,6 +9,20 @@
 
 
 def find_first_non_repeated_letter(str)
-  #write your method here
-  return true
+  i = 0
+
+  if str[i] != str[i+1]
+	  return str[0]
+  else
+	  while i < str.length
+		  if str[i] == str[i+1]
+			  i += 1
+		  elsif str[i+1] == str[i+2]
```

Comment:
If the first two characters in a given string are the same, the if statement in the while loop will find when they are no longer the same. Say str = "aaabbbbc". In this case the if statement would run through twice, but the third time the elsif statement would run because "a" != "b". "b" could be the first non-repeated letter, but there has to be a way to check if "b" is followed by another "b" or a different character. This is what the elsif block is for.  When the elsif block is commented out the method does not work properly. If str = "aaabbbbc",  without the elsif block, the letter "b" would be returned instead of "c".